qdb_dask_connector/__init__.py

```diff
@@ -0,0 +1,286 @@
+    # this function is used to mock part of future c api functionality
+    shard_size = conn.table(table_name.replace('"', "")).get_shard_size()
+    start, end = _extract_range_from_query(conn, query, table_name)
+    ranges_to_query = conn.split_query_range(start, end, shard_size)
+
+    subqueries = []
+    for rng in ranges_to_query:
+        subqueries.append(_create_subrange_query(query, rng))
+    return subqueries
+
+
+def _get_meta(conn, query: str, query_kwargs: dict) -> pd.DataFrame:
+    """
+    Returns empty dataframe with the expected schema of the query result.
+    """
+    np_res = conn.validate_query(query)
+    col_dtypes = {}
+    for id, column in enumerate(np_res):
+        col_dtypes[column[0]] = pd.Series(dtype=column[1].dtype)
+
+    df = pd.DataFrame(col_dtypes)
+    if query_kwargs["index"]:
+        df.set_index(query_kwargs["index"], inplace=True)
+    return df
+
+
+def _get_tasks_from_rest_api(
+    query: str, rest_api_uri: str
+) -> tuple[list[str], pd.DataFrame]:
+    raise NotImplementedError(
+        "REST API client mode is not implemented yet. Please use the Python API client mode."
+    )
+    # XXX:igor i used this code to validate that dask integration can work even when quasardb api is not installed
+    return [query], pd.DataFrame(
+        {
+            "$timestamp": pd.Series(dtype="datetime64[ns]"),
+            "$table": pd.Series(dtype="object"),
+            "x": pd.Series(dtype="float64"),
+        }
+    )
+
+
+def _get_tasks_from_python_client(
+    query: str, conn_kwargs: dict, query_kwargs: dict
+) -> tuple[list[str], pd.DataFrame]:
+    table_name = _extract_table_name_from_query(query)
+    with quasardb.Cluster(**conn_kwargs) as conn:
+        meta = _get_meta(conn, query, query_kwargs)
+        return _get_subqueries(conn, query, table_name), meta
+
+
+def query(
+    query: str,
+    *,
+    cluster_uri: str,
+    # rest api options
+    rest_uri: str = "",
+    # python api options
+    user_name: str = "",
+    user_private_key: str = "",
+    cluster_public_key: str = "",
+    timeout: datetime.timedelta = datetime.timedelta(seconds=60),
+    enable_encryption: bool = False,
+    client_max_parallelism: int = 0,
+    # query options
+    index: str | int = None,
+):
+    if not re.match(general_select_pattern, query):
+        raise NotImplementedError(
+            "Only SELECT queries are supported. Please refer to the documentation for more information."
+        )
+
+    conn_kwargs = {
+        "uri": cluster_uri,
+        "user_name": user_name,
+        "user_private_key": user_private_key,
+        "cluster_public_key": cluster_public_key,
+        "timeout": timeout,
+        "enable_encryption": enable_encryption,
+        "client_max_parallelism": client_max_parallelism,
+    }
+
+    query_kwargs = {
+        "index": index,
+    }
+
+    if rest_uri:
+        subqueries, meta = _get_tasks_from_rest_api(query, rest_uri)
```

Comment:
again, avoid calling this "subqueries", subqueries are a very different thing. just call them tasks.